interface/modules/zend_modules/module/Installer/src/Installer/Controller/InstallerController.php

```diff
@@ -327,4 +435,104 @@ public function nickNameAction()
+            return $version;
+        }
+        return false;
+    }
+
+    public function getFilesForUpgrade($modDirectory)
+    {
+        $ModulePath = $GLOBALS['srcdir']."/../".$GLOBALS['baseModDir']."zend_modules/module/".$modDirectory;
+        $sqldir = $ModulePath."/sql";
+        $versions = [];
+        $dh = opendir($sqldir);
+        if (!$dh) {
+            return false;
+        }
+
+        while (false !== ($sfname = readdir($dh))) {
+            if (substr($sfname, 0, 1) == '.') {
+                continue;
+            }
+
+            if (preg_match('/^(\d+)_(\d+)_(\d+)-to-\d+_\d+_\d+_upgrade.sql$/', $sfname, $matches)) {
+                $version = $matches[1] . '.' . $matches[2] . '.' . $matches[3];
+                $versions[$version] = $sfname;
+            }
+        }
+        $arrayKeys = array_keys($versions);
+        usort($arrayKeys, 'version_compare');
+        $sortVersions = array();
+        foreach ($arrayKeys as $key) {
+            $sortVersions[$key] = $versions[$key];
+        }
+        return $sortVersions;
+    }
+
+    public function makeButtonForSqlAction(InstModule $mod)
+    {
+        $dirModule = $this->getInstallerTable()->getRegistryEntry($mod->modId, "mod_directory");
+        $ModulePath = $GLOBALS['srcdir']."/../".$GLOBALS['baseModDir']."zend_modules/module/".$dirModule -> modDirectory;
+        $sqldir = $ModulePath."/sql";
+        $mod->sql_action = "";
+
+        if (file_exists($sqldir."/install.sql") && file_exists($ModulePath."/version.php") && empty($mod->sql_version)) {
+            $mod->sql_action = "install";
+        }
+
+        if (!empty($mod->sql_version) && $mod->sqlRun == 1) {
+            $versions = $this->getFilesForUpgrade($mod->modDirectory);
+
+            if (count($versions) > 0) {
+                foreach ($versions as $version => $sfname) {
+                    if (version_compare($version, $mod->sql_version) < 0) {
+                        continue;
+                    }
+                    $mod->sql_action = "upgrade";
+                }
+            }
+        }
+        return $mod;
+    }
+
+    public function makeButtonForACLAction(InstModule $mod)
+    {
+        $dirModule = $this->getInstallerTable()->getRegistryEntry($mod->modId, "mod_directory");
+        $ModulePath = $GLOBALS['srcdir']."/../".$GLOBALS['baseModDir']."zend_modules/module/".$dirModule -> modDirectory;
+        $sqldir = $ModulePath."/acl";
+        $mod->acl_action = "";
+
+        if (file_exists($sqldir."/acl_setup.php") && file_exists($ModulePath."/version.php") && empty($mod->acl_version)) {
+            $mod->acl_action = "install";
+        }
+        if (file_exists($sqldir."/acl_upgrade.php") && file_exists($ModulePath."/version.php") && !empty($mod->acl_version)) {
+            global $ACL_UPGRADE;
+            $unique_id = RandomGenUtils::createUniqueToken();
+            $_SESSION['lang_module_unique_id'] = $unique_id;
```

Comment:
Recommend changing the name to something pertinent such as:
```
$_SESSION['acl_upgrade_unique_id'] = $unique_id;
```

Reply:
Fixed